### PR TITLE
Apply decoupling of social media to the rest of the app

### PR DESCRIPTION
**Pull request by GabrielFleischer**
The PR is the second and final part of the decoupling of the social media feature.

It applies the new decoupled repository to the rest of the app

### CODE REVIEW

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java [hunk 1]
```diff
@@ -48,100 +36,68 @@ public class ChirpHandlerTest {
...
-    Lao updatedLao = laoView.createLaoCopy();
+    verify(socialMediaRepo).deleteChirp(LAO_ID, CHIRP_ID);
+    verifyNoMoreInteractions(socialMediaRepo);
+  }
 
-    Optional<Chirp> chirpOpt = updatedLao.getChirp(message.getMessageId());
-    assertTrue(chirpOpt.isPresent());
-    Chirp chirp = chirpOpt.get();
+  @Test
+  public void testUnorderedDeleteChirp() {
```
Comment (Warhedgehog) [inline fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java:92]: I don't understand the title of the test. Does it mean that the chirp is not present ?

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaHomeFragment.java [hunk 2]
```diff
@@ -64,28 +59,10 @@ private void setupSendButton() {
...
-          handler.postDelayed(
-              () -> {
-                if (swipeRefreshLayout.isRefreshing()) {
-                  swipeRefreshLayout.setRefreshing(false);
-                }
-              },
-              1000);
-        });
-  }
-
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaHomeFragment.java:84]: While I'm very glad that this type of updating can go :), I think you should remove the `swipeRefreshLayout` from `social_media_home_fragment` as well.

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Very, very good work ! In particular the change to tests is very good.
 A couple things I'd like changed though :)

I think the title should that it concerns SocialMedia. If for some reason someone want to find this PR once it's closed, with the title it will be hard.
Also the couple details in comments

**Comment by Warhedgehog** [comment]
I'll add that I tested the app and was unable to display chirps but it's the same on master

**Comment by GabrielFleischer** [comment]
> I'll add that I tested the app and was unable to display chirps but it's the same on master

Ah... that's annoying

File: fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java (hunk 1 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/test/unit/java/com/github/dedis/popstellar/utility/handler/ChirpHandlerTest.java:92]: I'll take a look at it and give a better name

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaHomeFragment.java (hunk 2 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaHomeFragment.java:84]: You're right 😄

**Comment by Warhedgehog** [comment]
> > I'll add that I tested the app and was unable to display chirps but it's the same on master
> 
> Ah... that's annoying

After thoughts the problem is not the same on master and on this branch I believe. On master, when updating (it's manual) it causes an exception when trying to display chirps. On this branch the automatic update is just not done, even though the backends receives it. I haven't had time to look into more details

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java [hunk 3]
```diff
@@ -72,7 +72,11 @@ private void sendNewChirp() {
                   null,
                   Instant.now().getEpochSecond())
               .subscribe(
-                  msg -> {},
+                  msg ->
+                      SocialMediaActivity.setCurrentFragment(
+                          getParentFragmentManager(),
+                          R.id.fragment_social_media_home,
+                          SocialMediaHomeFragment::newInstance),
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/socialmedia/SocialMediaSendFragment.java:79]: Good addition

File: fe2-android/app/src/main/res/layout/social_media_home_fragment.xml [hunk 4]
```diff
@@ -22,38 +22,14 @@
...
-        android:id="@+id/social_media_send_fragment_button"
-        android:layout_width="@dimen/round_button_size"
-        android:layout_height="@dimen/round_button_size"
-        android:layout_alignParentEnd="true"
-        android:layout_alignParentBottom="true"
-        android:layout_marginEnd="@dimen/margin_top"
-        android:layout_marginBottom="@dimen/margin_top"
-        android:background="@drawable/round_button"
-        android:contentDescription="@string/send_button_desc"
-        android:src="@drawable/ic_social_media_send_foreground" />
```
Comment (Warhedgehog) [inline fe2-android/app/src/main/res/layout/social_media_home_fragment.xml:56]: You should not have deleted that I think. This causes the build to fail.

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Nice improvements. Sadly the build fails :(

File: fe2-android/app/src/main/res/layout/social_media_home_fragment.xml (hunk 4 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/main/res/layout/social_media_home_fragment.xml]: Fixed

**Review by Warhedgehog — APPROVED** [review]
LGTM

File: fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/DataRegistryModuleHelper.java [hunk 5]
```diff
@@ -16,19 +16,38 @@ public static DataRegistry buildRegistry() {
...
       MessageRepository msgRepo,
       KeyManager keyManager,
       ServerRepository serverRepo) {
     LaoHandler laoHandler = new LaoHandler(keyManager, msgRepo, laoRepo, serverRepo);
     RollCallHandler rollCallHandler = new RollCallHandler(keyManager, laoRepo);
     ElectionHandler electionHandler = new ElectionHandler(msgRepo, laoRepo);
     ConsensusHandler consensusHandler = new ConsensusHandler(laoRepo);
-    ChirpHandler chirpHandler = new ChirpHandler(laoRepo);
+    ChirpHandler chirpHandler = new ChirpHandler(laoRepo, socialMediaRepo);
     TransactionCoinHandler transactionCoinHandler = new TransactionCoinHandler(laoRepo);
```
Comment (pierluca) [inline fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/DataRegistryModuleHelper.java:51]: I'm a bit concerned that when we'll be done with the decoupling, this static method will have a huge parameter list...

**Review by pierluca — APPROVED** [review]

File: fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/DataRegistryModuleHelper.java (hunk 5 above)
Comment (GabrielFleischer) [reply fe2-android/app/src/test/framework/robolectric/java/com/github/dedis/popstellar/di/DataRegistryModuleHelper.java:51]: That's exactly the concern I raised in #1210. It is better, but far from perfect.

I don't have any good solution right now too.
Maybe a builder with a default value for each repository.

**Pull request merged**